README.md

```diff
@@ -1,6 +1,26 @@
 # Kubectl
 
+[![Build Status]](https://travis-ci.org/kubernetes/kubectlw) [![GoDoc](https://godoc.org/k8s.io/kubectl?status.svg)](https://godoc.org/k8s.io/kubectl)
+
 The k8s.io/kubectl repo is used to track issues for the kubectl cli distributed with k8s.io/kubernetes.
+It also contains libraries used by client programs (mostly `kubectl`). This is currently vendored into `k8s.io/kubernetes`.
+
+# Style guide
+
+Here is a list of guidelines to follow when submitting code to that repository:
+
+- Full unit-test coverage.
+
+- Go tools compliant (`go get`, `go test`, etc.). It needs to be vendorable
+  somewhere else.
+
+- No dependence on `k8s.io/kubernetes`. This repository can depend on other
+  sub-repositories.
+
+- Libraries must be documented properly.
```

Comment:
```
Code must be usefully [commented](https://golang.org/doc/effective_go.html#commentary).
```
The repository's [pre-commit](https://github.com/kubernetes/kubectl/blob/master/bin/pre-commit.sh) script enforces golint and go vet which check certain comment related things.